hotsos/core/plugins/kernel/kernlog/calltrace.py

```diff
@@ -206,6 +206,58 @@ def fields(self):
                 'local_pcp', 'free_cma']
 
 
+class BcacheTraceType(TraceTypeBase):
+
+    def __init__(self):
+        self._search_def = None
+        self.generics = []
+
+    @property
+    def name(self):
+        return 'calltrace-bcache'
+
+    @property
+    def searchdef(self):
+        if self._search_def:
+            return self._search_def
+
+        start = SearchDef(r'{} (\S+) task bcache-register:(d+)'
+                          r' blocked for more than (d+) seconds.'.
+                          format(KERNLOG_PREFIX))
+        body = SearchDef(r'(\S+) schedule(\S+) closure_sync(\S+)'
+                         r' bch_journal_meta(\S+) bch_btree_set_root(\S+)'
```

Comment:
Given that you are not extracting any information from results of this search is it possible to reduce the size of this expression? Remember that the longer the search expression the more CPU cycles it will take to run it and if you are storing items from the result this will also consume memory. Try to reduce the search to its absolute minimum.

Reply:
I have reduced the search terms but I need to keep some in there to not run the chance of hitting some other bcache trace and confusing it for this deadlock